src/main/java/seedu/toluist/controller/FindController.java

```diff
@@ -0,0 +1,151 @@
+package seedu.toluist.controller;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.logging.Logger;
+
+import seedu.toluist.commons.core.LogsCenter;
+import seedu.toluist.dispatcher.CommandResult;
+import seedu.toluist.model.Task;
+import seedu.toluist.model.TodoList;
+import seedu.toluist.ui.Ui;
+
+/**
+ * Searches the task list for matches in the parameters, and displays the results received
+ */
+public class FindController extends Controller {
+    private static final String COMMAND_TEMPLATE = "^(?<command>(find|filter|list))"
+            + "(\\s+(?<keywords>.+))?\\s*";
+
+    private static final String TAG_PARAMETER = "tag/";
+    private static final String NAME_PARAMETER = "name/";
+    private static final String TRUE_PARAMETER = "true";
+    private static final String FALSE_PARAMETER = "false";
+    private static final String KEYWORDS_PARAMETER = "keywords";
+
+    private static final int NUMBER_OF_SPLITS_FOR_COMMAND_PARSE = 2;
+    private static final String COMMAND_SPLITTER_REGEX = " ";
+    private static final int PARAMETER_SECTION = 1;
+
+    private static final String RESULT_MESSAGE_TEMPLATE = "Searching for \"%s\" by %s.\n%d results found";
+    private static final String NAME_MESSAGE = "name";
+    private static final String TAG_MESSAGE = "tag";
+    private static final String NAME_AND_TAG_MESSAGE = "name and tag";
+
+    private final Logger logger = LogsCenter.getLogger(getClass());
+
+    public FindController(Ui renderer) {
+        super(renderer);
+    }
+
+    public CommandResult execute(String command) {
+        logger.info(getClass() + "will handle command");
+
+        //initialize keywords and variables for searching
+        HashMap<String, String> tokens = tokenize(command);
+        boolean isSearchByTag = tokens.get(TAG_PARAMETER).equals(TRUE_PARAMETER);
+        boolean isSearchByName = tokens.get(NAME_PARAMETER).equals(TRUE_PARAMETER);
+        String[] keywordList = convertToArray(tokens.get(KEYWORDS_PARAMETER));
+
+        //initialize search parameters and variables
+        ArrayList<Task> foundTasks = new ArrayList<Task>();
+        TodoList todoList = TodoList.load();
+
+        for (int j = 0; j < keywordList.length; j++) {
+            if (isSearchByTag) {
+                addTasksIgnoreDuplicate(foundTasks, todoList.getTasksWhosTagsContains(keywordList[j]));
```

Comment:
i changed it to set. it works still, but now it doesn't follow the default order inputted. I guess this doesn't matter right now because we haven't done our sorting methods yet, so order doesn't matter?